widgets/treeview/treeview.go

```diff
@@ -0,0 +1,818 @@
+package treeview
+
+import (
+	"errors"
+	"fmt"
+	"image"
+	"io"
+	"log"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/mum4k/termdash/cell"
+	"github.com/mum4k/termdash/keyboard"
+	"github.com/mum4k/termdash/mouse"
+	"github.com/mum4k/termdash/private/canvas"
+	"github.com/mum4k/termdash/private/runewidth"
+	"github.com/mum4k/termdash/terminal/terminalapi"
+	"github.com/mum4k/termdash/widgetapi"
+)
+
+// Number of nodes to scroll per mouse wheel event
+const (
+	ScrollStep = 5
+)
+
+// TreeNode represents a node in the treeview.
+type TreeNode struct {
+	ID            string
```

Comment:
Would you mind documenting all these attributes? It would help the posterity to maintain this widget.